test/MockAlarmClock.h

```diff
@@ -0,0 +1,149 @@
+/* 
+ * File:   MockAlarmClock.h
+ * Author: Craig Cogdill
+ * Created: November 3, 2015 2pm
+ */
+
+#pragma once
+#include <iostream>
+#include <atomic>
+
+
+template<typename Duration> class MockAlarmClock {
```

Comment:
A mock class has to inherit from the class being tested. Mostly you want to test the base class functionality with select methods being overridden for the purpose of forcing different branching in the rest of the class. Since you are using a class template and not a real class this becomes more difficult. Check out: http://stackoverflow.com/questions/8810224/inheriting-from-a-template-class-in-c